packages/pwa/app/utils/site-utils.test.js

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {resolveSiteFromUrl} from './site-utils'
+import {setConfig} from 'pwa-kit-react-sdk/ssr/universal/utils'
+
+import mockConfig from '../../config/mocks/default'
+
+beforeEach(() => {
+    jest.resetModules()
+})
+
+describe('resolveSiteFromUrl', function() {
+    test('throw an error without an arg', () => {
+        expect(() => {
+            resolveSiteFromUrl()
+        }).toThrow()
+    })
+
+    test('return site based on the site alias in the url', () => {
+        const result = resolveSiteFromUrl('https://www.example-site.com/us/en-US/women/dress')
+        expect(result.id).toEqual('site-2')
+    })
+
+    test('return default site for home page', () => {
+        const result = resolveSiteFromUrl('https://www.example-site.com/')
+        expect(result.id).toEqual('site-1')
+    })
+
+    test('throw an error when no site can be found', () => {
```

Comment:
```suggestion
    test('throw an error when no matching default site can be found', () => {
```

I was a bit confused initially: I saw there are sites in the config. But actually we're testing for when there isn't `site-3` found in the `sites` config.

Reply:
This case is to test out the case when someone inputs a URL with a random site, since we mostly resolve the site from url, we want to make sure it throws an error when there is not matching site